cmd/addon-operator-manager/main.go

```diff
@@ -187,16 +187,18 @@ func setup() error {
 		if !apierrors.IsNotFound(err) {
 			return fmt.Errorf("failed to GET the AddonOperator object: %w", err)
 		}
+		// TODO: is it correct that we just ignore the error and reset the ado object
 		addonOperatorObjectInCluster = addonsv1alpha1.AddonOperator{}
```

Comment:
If you notice carefully, this code would execute as a consequence of  `apierrors.IsNotFound(err) == true` for the AddonOperator CR which is (and has been since the past) fine with ADO i.e. sometimes addon-operator spins up before AddonOperator CR appears in the cluster via the `handleAddonOperatorCreation` method :)

Reply:
Oh got it, thanks!